bum/src/index.js

```diff
@@ -0,0 +1,86 @@
+const core = require('@actions/core');
+const fg = require('fast-glob');
+const { run } = require('../../utils');
+const loadGroupTypeDefinition = require('./grouptype-definition');
+const configureGroupTypes = require('./configure-grouptypes');
+const { loadCredentials } = require('../../utils');
+const fetchIamToken = require('../../iam-test-token/src/iam-auth');
+const setupGcloud = require('../../setup-gcloud/src/setup-gcloud');
+const projectInfo = require('../../cloud-run/src/project-info');
+
+const gcloudAuth = async (serviceAccountKey) => setupGcloud(
+  serviceAccountKey,
+  process.env.GCLOUD_INSTALLED_VERSION || 'latest',
+);
+
+const setupEnvironment = async (
+  serviceAccountKey, gcloudAuthKey, groupTypes, bumUrl,
+) => {
+  const projectId = await gcloudAuth(gcloudAuthKey);
+  const { env: projectEnv } = projectInfo(projectId);
+
+  // Skip BUM handling if staging (not business-unit-api)
+  let skipBum = (projectEnv === 'staging');
+  // By default we always target prod business-unit-api unless explicitly told not to.
+  let credentialsEnv = 'prod';
+  let url = bumUrl;
+  if (projectEnv === 'staging' && bumUrl.endsWith('retailsvc.dev')) {
+    credentialsEnv = 'staging';
+    skipBum = false;
+    core.warning(`GroupTypes definitions has been explicitly configured to publish to ${bumUrl} which is the STAGING environment.`);
+  } else if (projectEnv === 'prod' && bumUrl.endsWith('retailsvc.dev')) {
+    url = bumUrl.replace('dev', 'com');
+  }
+
+  const {
+    iamApiEmail,
+    iamApiPassword,
+    iamApiKey,
+    iamApiTenant,
+  } = await loadCredentials(serviceAccountKey, credentialsEnv);
+  let iamToken = '';
+  if (!skipBum) {
+    iamToken = await fetchIamToken(iamApiKey, iamApiEmail, iamApiPassword, iamApiTenant);
```

Comment:
How do you ensure this token has the permissions to publish to BUM?

Reply:
Do I need to do that, if it fails it will fail the pipeline for the service and that service need to ensure that they use a token with permission.